lib/libutils/isoc/include/assert.h

```diff
@@ -31,3 +31,5 @@ void _assert_log(const char *expr, const char *file, const int line,
 	} while (0)
 
 #endif
+
+#define static_assert _Static_assert
```

Comment:
Should `COMPILE_TIME_ASSERT` be deprecated? I think it should. `static_assert` is better because it can be used in some places where `COMPILE_TIME_ASSERT` cannot, and it is standard. One drawback is the C11 version requires a string argument, but the C2x version doesn't. GCC doesn't seem to care but Clang in `-std=gnu99` mode emits a warning:

```
$ clang -std=gnu99 -o test test.c
test.c:7:20: warning: '_Static_assert' with no message is a C2x extension [-Wc2x-extensions]
static_assert(0 < 1);
                   ^
                   , ""
1 warning generated.

```

So, how about adding this:
```diff
diff --git a/mk/compile.mk b/mk/compile.mk
index b5ab62b0..51ba074c 100644
--- a/mk/compile.mk
+++ b/mk/compile.mk
@@ -35,7 +35,8 @@ comp-cflags-warns-high = \
        -Wmissing-prototypes -Wnested-externs -Wpointer-arith \
        -Wshadow -Wstrict-prototypes -Wswitch-default \
        -Wwrite-strings \
-       -Wno-missing-field-initializers -Wno-format-zero-length
+       -Wno-missing-field-initializers -Wno-format-zero-length \
+       -Wno-c2x-extensions \
 ifeq ($(CFG_WARN_DECL_AFTER_STATEMENT),y)
 comp-cflags-warns-high += $(call cc-option,-Wdeclaration-after-statement)
 endif
```

...as well as a comment near `COMPILE_TIME_ASSERT` to mention that `static_assert` should be used instead?

Reply:
Hmm, it seems that GCC 8.2 acts funny when omitting the message to `static_assert`.
```
core/arch/arm/kernel/thread_optee_smc.c:31:50: error: expected ‘,’ before ‘)’ token
        OPTEE_SMC_ASYNC_NOTIF_VALUE_DO_BOTTOM_HALF);
                                                  ^
cc1: error: unrecognized command line option ‘-Wno-c2x-extensions’ [-Werror]
```

I suppose it can be worked around with some macro trickery, but is it worth it? I'll try to see how hard it is.

Reply:
Oh, perhaps we need `$(call cc-option,-Wno-c2x-extensions)` instead of plain `-Wno-c2x-extensions`.

Reply:
> Oh, perhaps we need `$(call cc-option,-Wno-c2x-extensions)` instead of plain `-Wno-c2x-extensions`.

I'm not sure that helps, the message seems to be shown only when the static_assert is fails.